noria-server/mir/src/optimize.rs

```diff
@@ -149,29 +140,23 @@ fn find_and_merge_filter_aggregates(q: &mut MirQuery) -> Vec<MirNodeRef> {
             child.from_version,
             child.columns.clone(),
             MirNodeType::FilterAggregation {
-                on: match &agg {
-                    MirNodeType::Aggregation { on, group_by: _, kind: _ } => on.clone(),
-                    _ => unreachable!(),
-                },
+                on: if let MirNodeType::Aggregation { on, .. } = &agg {
```

Comment:
Looking at this again, maybe the `if let` should just go on the outside of this constructor instead? Then you can match on all the fields at once, and just have one `unimplemented!` clause.